app/src/stores/fields.ts

```diff
@@ -65,308 +66,338 @@ const fakeFilesField: Field = {
-				if (
-					(relation.collection === collectionKey && relation.field === fieldKey) ||
-					(relation.related_collection === collectionKey && relation.meta?.one_field === fieldKey)
-				) {
-					return false;
-				}
+	/**
+	 * Retrieve field info for a field or a related field
+	 */
+	function getField(collection: string, fieldKey: string): Field | null {
+		if (fieldKey.includes('.')) {
+			return getRelationalField(collection, fieldKey) || null;
+		} else {
+			return fields.value.find((field) => field.collection === collection && field.field === fieldKey) || null;
+		}
+	}
 
-				return true;
-			});
+	/**
+	 * Retrieve nested fields for a given group field
+	 */
+	function getFieldGroupChildren(collection: string, fieldKey: string): Field[] | null {
+		return fields.value.filter((field) => field.collection === collection && field.meta?.group === fieldKey) || null;
+	}
 
-			try {
-				await api.delete(`/fields/${collectionKey}/${fieldKey}`);
-				await collectionsStore.hydrate();
-			} catch (err: any) {
-				this.fields = stateClone;
-				relationsStore.relations = relationsStateClone;
-				unexpectedError(err);
-			}
-		},
-		getPrimaryKeyFieldForCollection(collection: string): Field | null {
-			const primaryKeyField = this.fields.find(
-				(field) => field.collection === collection && field.schema?.is_primary_key === true
-			);
-
-			return primaryKeyField ?? null;
-		},
-		getFieldsForCollection(collection: string): Field[] {
-			return orderBy(
-				this.fields.filter((field) => field.collection === collection),
-				[(field) => field.meta?.system === true, (field) => (field.meta?.sort ? Number(field.meta?.sort) : null)],
-				['desc', 'asc']
-			);
-		},
-		getFieldsForCollectionAlphabetical(collection: string): Field[] {
-			return this.getFieldsForCollection(collection).sort((a: Field, b: Field) => {
-				if (a.field < b.field) return -1;
-				else if (a.field > b.field) return 1;
-				else return 1;
-			});
-		},
-		/**
-		 * Retrieve sorted fields including groups. This is necessary because
-		 * fields inside groups starts their sort number from 1 to N again.
-		 */
-		getFieldsForCollectionSorted(collection: string): Field[] {
-			const fields = orderBy(
-				this.fields.filter((field) => field.collection === collection),
-				'meta.sort'
-			);
-
-			const nonGroupFields = fields.filter((field: Field) => !field.meta?.group);
-
-			for (const [index, field] of nonGroupFields.entries()) {
-				const groupFields = fields.filter((groupField: Field) => groupField.meta?.group === field.field);
-
-				if (groupFields.length) {
-					nonGroupFields.splice(index + 1, 0, ...orderBy(groupFields, 'meta.sort'));
-				}
-			}
+	/**
+	 * Retrieve field info for a (deeply) nested field
+	 * Recursively searches through the relationships to find the field info that matches the
+	 * dot notation.
+	 */
+	function getRelationalField(collection: string, fields: string) {
```

Comment:
Does this function not need to be exposed?

Reply:
It isn't used anywhere except inside this store as far as I can see.